rules/CodingStyle/Rector/ClassConst/RemoveFinalFromConstRector.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\ClassConst;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassConst;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Privatization\NodeManipulator\VisibilityManipulator;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://php.watch/versions/8.1/final-class-const
+ *
+ * @see \Rector\Tests\CodingStyle\Rector\ClassConst\RemoveFinalFromConstRector\RemoveFinalFromConstRectorTest
+ */
+final class RemoveFinalFromConstRector extends AbstractRector
+{
+    public function __construct(
+        private readonly VisibilityManipulator $visibilityManipulator
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Remove final from constants in classes defined as final', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    final public const NAME = 'value';
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public const NAME = 'value';
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassConst::class];
+    }
+
+    /**
+     * @param ClassConst $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $parentClass = $this->betterNodeFinder->findParentType($node, Class_::class);
+
+        if (! $parentClass instanceof Class_) {
+            return null;
+        }
+
+        if ($parentClass->isFinal() && $node->isFinal()) {
+            $this->visibilityManipulator->removeFinal($node);
+        }
+
+        return $node;
```

Comment:
can return `$node` early, so next will return `null` as not changed.

```suggestion
            $this->visibilityManipulator->removeFinal($node);
            return $node;
        }
        
        return null;    
```